src/java.base/share/classes/jdk/internal/foreign/MemorySessionImpl.java

```diff
@@ -60,7 +59,8 @@
     static final VarHandle STATE;
     static final int MAX_FORKS = Integer.MAX_VALUE;
 
-    public static final MemorySessionImpl GLOBAL = new GlobalSession(null);
+    public static final MemorySessionImpl GLOBAL = new GlobalSession(null, false);
+    public static final MemorySessionImpl NATIVE = new GlobalSession(null, true);
```

Comment:
We no longer need `isInternal`. But we do need, I think, another session for native segments returned by linker, so that comparisons against global arena's scope would fail.

Reply:
Ended up (after offline discussion) with a simpler implementation. A new session is created every time for the externally managed segments. The working theory is that scalarization should still work well (because a global session doesn't have a resource list - it's a pretty small and dumb object).

We will go for the simple version, test its performance - and then decide if it's good enough. If performance is not good and cannot be improved, we might consider an API solution to the problem (where scope accessors are optional). But we need to gather some numbers first.